analyses/cell-type-ewings/cell_type_annotations/SCPCS000490/SCPCL000822_tumor-cell-validation.Rmd

```diff
@@ -0,0 +1,492 @@
+coldata_df |> 
+  dplyr::select(UMAP1, UMAP2, barcodes, combined_cnv_classification) |>
+  unique() |> 
+  ggplot(aes(x = UMAP1, y = UMAP2, color = combined_cnv_classification)) +
+  geom_point(alpha = 0.5, size = 0.5)
+```
+
+The remainder of the notebook will focus on validating that these in fact are tumor cells and all other cells are classified as normal. 
+
+## Validate tumor cells 
+
+### Marker gene expression 
+
+First, we will look at the expression of the tumor marker genes. 
+We expect that these should be expressed higher in the tumor cells than in the normal cells. 
+
+In this first plot, we show a density plot of marker gene expression coloring by the cell classification (tumor or normal). 
+For this plot, we will include all classifications, including the new combined CNV classification. 
+
+```{r}
+# calculate sum of the scaled gene expression values for plotting 
+marker_density_df <- coldata_df |> 
+  dplyr::select(barcodes, gene_symbol, gene_expression, ends_with("classification")) |> 
+  unique() |> 
+  dplyr::group_by(gene_symbol) |>
+  # get z-scores for each gene
+  dplyr::mutate(transformed_gene_expression = scale(gene_expression)[, 1]) |> 
+  dplyr::ungroup() |> 
+  dplyr::group_by(barcodes) |> 
+  # add all the scores together 
+  dplyr::mutate(sum_exp = sum(transformed_gene_expression)) |>
+  # expand classifications column for easy faceting 
+  dplyr::ungroup() |> 
+  tidyr::pivot_longer(
+    cols = ends_with("classification"),
+    names_to = "method",
+    values_to = "classification"
+  )
+
+ggplot(marker_density_df, aes(x = sum_exp, color = classification)) +
+  geom_density() +
+  facet_wrap(vars(method))
+```
+
+This plot shows us that overall the tumor cells in each method have higher marker gene expression. 
+Although the `InferCNV` alone and combined CNV classifications appear to have the most distinct separation in marker gene expression. 
+
+Next, we will plot the expression of each gene in each cell using a heatmap. 
+Here the rows are the marker genes and the columns are the cells with the color indicating the expression of that gene in that cell. 
+
+```{r}
+# create matrix with marker genes as rows and barcodes as columns
+marker_gene_heatmap <- coldata_df |> 
+  dplyr::select(gene_expression, gene_symbol, barcodes) |> 
+  tidyr::pivot_wider(values_from = gene_expression, 
+                     names_from = barcodes) |> 
+  tibble::column_to_rownames("gene_symbol") |> 
+  as.matrix()
+
+# create annotation df, make sure that dimensions match up by only getting unique barcodes 
+annotation_df <- coldata_df |> 
+  dplyr::select(barcodes, combined_cnv_classification) |> 
+  unique() 
+
+annotation <- ComplexHeatmap::columnAnnotation(
+  classification = annotation_df$combined_cnv_classification,
+  col = list(classification = c("Tumor" = "#00274C", "Normal" = "#FFCB05")))
+```
+
+
+```{r}
+# plot heatmap of marker genes 
+ComplexHeatmap::Heatmap(
+  marker_gene_heatmap,
```

Comment:
Just for clarity: this plot end up showing scaled expression somehow? The label is `matrix_29` and it looks like almost all values are at 1 or 0?